### PR TITLE
Feature/chris 5/upload profile photo

### PR DESCRIPTION
**Pull request by ChrisTangSdy**
Finshed upload file using amazon s3:

1. set up amazon s3
2. set up multer
3. Logic upload profile image to server, then to amazon s3 , then delete server profile image.
4. when fetching from frontend, frontend will get an link to access image file

### CODE REVIEW

File: src/controllers/profile.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const createProfile = async (req, res) => {
     instagram,
     linkedin,
   } = req.body;
-  console.log(skills)
+  console.log(skills);
```
Comment (Ollie-lee) [inline src/controllers/profile.js]: remove

File: src/controllers/profile.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ const createProfile = async (req, res) => {
   if (facebook) profileFields.social.facebook = facebook;
   if (linkedin) profileFields.social.linkedin = linkedin;
   if (instagram) profileFields.social.instagram = instagram;
-console.log(profileFields)
+  console.log(profileFields);
   try {
```
Comment (Ollie-lee) [inline src/controllers/profile.js:68]: remove

Comment (ChrisTangSdy) [reply src/controllers/profile.js:68]: removed

File: src/controllers/profile.js (hunk 1 above)
Comment (ChrisTangSdy) [reply src/controllers/profile.js]: removed

File: src/controllers/user.js [hunk 3]
```diff
@@ -49,11 +55,80 @@ const getUserById = async (req, res) => {
...
+    res.status(500).send('Server Error');
+  }
+};
+
+const getUserPhoto = async (req, res) => {
+  try {
+    const user = await User.findById(req.user.id);
+    // let getPath = path.join(__dirname, `/../uploads/${user.photo}`);
+
+    res.sendFile(
```
Comment (Ollie-lee) [inline src/controllers/user.js]: 发送的应该是url到前端,这个url是该文件在后端被host的地址

File: src/controllers/user.js [hunk 4]
```diff
@@ -49,11 +55,80 @@ const getUserById = async (req, res) => {
...
+        }
+      }
+    );
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server Error');
+  }
+};
+
+const uploadUserPhoto = async (req, res) => {
```
Comment (Ollie-lee) [inline src/controllers/user.js:62]: 这个api和62行的api是不是用处一样？这个是新的，62行的是没用的？

File: src/controllers/user.js (hunk 3 above)
Comment (ChrisTangSdy) [reply src/controllers/user.js]: That's a useless function that should be deleted. Sorry,  I forget to delete

File: src/controllers/user.js (hunk 4 above)
Comment (ChrisTangSdy) [reply src/controllers/user.js:62]: Yep, the codes between 62-97 are useless. I forget to delete them

Comment (ChrisTangSdy) [reply src/controllers/user.js:62]: Already delete useless funtcion

**Review by Ollie-lee — APPROVED** [review]

**Pull request merged**